src/App.tsx

```diff
@@ -16,47 +16,93 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [
+    sortedGoods,
+    setSortedGoods,
+  ] = useState(goodsFromServer);
+  const [isVisibleReser, setIsVisibleReser] = useState(false);
+
+  enum SortType {
+    Alphabet = 'alphabet',
+    Length = 'length',
+    Reverse = 'reverse',
+  }
+
+  function handleSort(filter: string) {
+    let sorted: string[] = [];
+
+    if (filter === SortType.Alphabet) {
+      sorted = [...sortedGoods].sort();
+    }
+
+    if (filter === SortType.Length) {
+      sorted = [...sortedGoods].sort((a, b) => a.length - b.length);
+    }
+
+    if (filter === SortType.Reverse) {
+      sorted = [...sortedGoods].reverse();
+    }
+
+    setSortedGoods(sorted);
+    setIsVisibleReser(true);
+  }
```

Comment:
This function should be moved outside the component definition.

Reply:
Not fixed

Reply:
can you help and explain how to do it? because I dont understand. Thanks

Reply:
Sorry, looks, like it was fixed, but better to use `switch/case` here